exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/FunctionProvider.kt

```diff
@@ -438,6 +439,60 @@ abstract class FunctionProvider {
         return "INSERT INTO ${transaction.identity(table)} $columnsExpr $valuesExpr"
     }
 
+    open fun merge(
+        dest: Table,
+        source: Table,
+        transaction: Transaction,
+        whenBranches: List<MergeBaseStatement.WhenBranchData>,
```

Comment:
It looks like `MergeBaseStatement.WhenBranchData` is not readable as a standalone. It should always be used together and meaningful when reading.

What do you think about `MergeStatement.WithWhen` or just `MergeStatement.When`?

@bog-walk, what do you think?

Reply:
It's quite often referenced as `clause` in the documentations. So another options might be `Clause` or `WhenClause`

Reply:
@e5l @obabichevjb `MergeStatement.When` and `MergeStatement.WhenClause` to me seem the most descriptive and to the point when used alone.
If we also want to keep it inline with the `MergeWhenAction` enum passed to it, for example, then simply `MergeStatement.When` would work well.

Reply:
Updated MR, removed any usages of the term "branch" and replaced it with the term "clause".

After 2 iterations I come up with the variant of usage only term "clause(s)" instead of "when"/"whenClauses". It may look more descriptive and simpler in the context of merge command, because there is only one term. With that variant interface of `FunctionProvider.merge()` looks like:

```kotlin
open fun merge(
    dest: Table,
    source: Table,
    transaction: Transaction,
    clauses: List<MergeStatement.Clause>,
    on: Op<Boolean>?
)
```

And the Clause data class:
```
data class Clause(
  val action: ClauseAction,
  val arguments: List<Pair<Column<*>, Any?>>,
  val and: Op<Boolean>?,
  /** ... */
  val deleteWhere: Op<Boolean>? = null
)

enum class ClauseAction {
  INSERT, UPDATE, DELETE
}
```

Minor but also `action` parameter of `Clause` class moved to the first position, as it defines the meaning of the following arguments.

What do you think about such a variant?

One commit before there is an option with `MergeStatement.When` data class, but I found out that it looks quite weird for the list parameters to call them `whens` or `whenClauses`